ui/src/views/WorkerManager/WMEditWorkerPool/index.jsx

```diff
@@ -0,0 +1,203 @@
+import MenuItem from '@material-ui/core/MenuItem/MenuItem';
+import CodeEditor from '@mozilla-frontend-infra/components/CodeEditor';
+import ListItem from '@material-ui/core/ListItem/ListItem';
+import Switch from '@material-ui/core/Switch/Switch';
+import { withStyles } from '@material-ui/core';
+import CheckIcon from 'mdi-react/CheckIcon';
+import List from '../../Documentation/components/List';
+import isWorkerTypeNameValid from '../../../utils/isWorkerTypeNameValid';
+import Dashboard from '../../../components/Dashboard';
+import Button from '../../../components/Button';
+
+const gcpConfig = {
+  minCapacity: 0,
+  maxCapacity: 0,
+  capacityPerInstance: 1,
+  machineType: 'n1-highcpu-8',
+  regions: ['us-west2'],
+  userData: {},
+  scheduling: {},
+  networkInterfaces: [{}],
+  disks: [{}],
+};
+
+@hot(module)
+@withStyles(theme => ({
+  successIcon: {
+    ...theme.mixins.successIcon,
+  },
+  createIconSpan: {
+    ...theme.mixins.fab,
+    ...theme.mixins.actionButton,
+    right: theme.spacing.unit * 11,
+  },
+}))
+export default class WMWorkerPoolEditor extends Component {
+  static defaultProps = {
+    isNewWorkerPool: true,
+  };
+
+  static propTypes = {
+    isNewWorkerPool: bool,
+  };
+
+  state = {
+    workerPool: {
+      name: '',
+      description: '',
+      owner: '',
+      wantsEmail: false,
+      providerType: 'gcp',
+      providerId: '',
+      config: gcpConfig,
+    },
+    invalidProviderConfig: false,
+  };
+
+  handleInputChange = ({ target: { name, value } }) => {
+    this.setState({ workerPool: { ...this.state.workerPool, [name]: value } });
+  };
+
+  handleSwitchChange = event => {
+    const {
+      target: { value },
+    } = event;
+
+    this.setState({
+      workerPool: {
+        ...this.state.workerPool,
+        [value]: !this.state.workerPool[value],
+      },
+    });
+  };
+
+  handleEditorChange = value => {
+    const { workerPool } = this.state;
+
+    try {
+      workerPool.config = JSON.parse(value);
+
+      this.setState({
+        workerPool,
+        invalidProviderConfig: false,
+      });
+    } catch (err) {
+      workerPool.config = value;
+
+      this.setState({
+        workerPool,
+        invalidProviderConfig: true,
+      });
+    }
+  };
+
+  render() {
+    const { isNewWorkerPool, classes } = this.props;
+    const { workerPool, invalidProviderConfig } = this.state;
+
+    return (
+      <Dashboard
+        title={
+          isNewWorkerPool
+            ? 'Worker Manager: Create Worker Type'
+            : 'Worker Manager: Edit Worker Type'
+        }>
+        <TextField
+          label="Enter Worker Type Name..."
+          name="name"
+          error={
+            Boolean(workerPool.name) && !isWorkerTypeNameValid(workerPool.name)
+          }
+          onChange={this.handleInputChange}
+          fullWidth
+          value={workerPool.name}
+          margin="normal"
+        />
+
+        <TextField
```

Comment:
Let's wrap the form in a `List` component and have individual form entries wrapped in a `ListItem`. Something similar to https://github.com/taskcluster/taskcluster/blob/master/ui/src/components/HookForm/index.jsx.